### PR TITLE
minor: plumb `context` through to tlog requests

### PR DESCRIPTION
**Pull request by dekkagaijin**
This makes the `timeout` argument unnecessary

```release-note
NONE
```

### CODE REVIEW

File: cmd/cosign/cli/attest/attest.go [hunk 1]
```diff
@@ -67,6 +67,12 @@ func AttestCmd(ctx context.Context, ko sign.KeyOpts, regOpts options.RegistryOpt
 		return errors.Wrap(err, "parsing reference")
 	}
 
+	if timeout != time.Duration(0) {
```
Comment (mattmoor) [inline cmd/cosign/cli/attest/attest.go:70]: This reads a lot better
```suggestion
	if timeout != 0 {
```

File: cmd/cosign/cli/sign/sign.go [hunk 2]
```diff
@@ -135,6 +134,13 @@ func SignCmd(ctx context.Context, ko KeyOpts, regOpts options.RegistryOptions, a
 		}
 	}
 
+	// TODO - Defaulting the timeout to zero as the CLI doesn't accept timeout.
```
Comment (mattmoor) [inline cmd/cosign/cli/sign/sign.go:137]: IDK how to interpret the work needed to address this TODO

Comment (dekkagaijin) [reply cmd/cosign/cli/sign/sign.go]: fixed it up

File: cmd/cosign/cli/attest/attest.go (hunk 1 above)
Comment (dekkagaijin) [reply cmd/cosign/cli/attest/attest.go]: done

**Review by mattmoor — APPROVED** [review]

**Pull request merged**